pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -915,49 +916,30 @@ func TestGarbageCollectorSync(t *testing.T) {
 
 	// Wait until the sync discovers the initial resources
 	time.Sleep(1 * time.Second)
-
-	err = expectSyncNotBlocked(fakeDiscoveryClient, &gc.workerLock)
```

Comment:
expectSyncNotBlocked seemed redundant now that the worker lock is removed. I think asserting the monitors should be good enough validation that gc.Sync is running and not blocked

Reply:
I tend to agree. That said, expectSyncNotBlocked always slept for 1 second, so make sure dropping that didn't introduce timing flakes (run the unit test with `stress` with and without -race detection on)